src/azure-cli-core/azure/cli/core/extension/operations.py

```diff
@@ -465,8 +465,10 @@ def list_available_extensions(index_url=None, show_details=False, cli_ctx=None):
             'name': name,
             'version': latest['metadata']['version'],
             'summary': latest['metadata']['summary'],
-            'preview': latest['metadata'].get(EXT_METADATA_ISPREVIEW, False),
-            'experimental': latest['metadata'].get(EXT_METADATA_ISEXPERIMENTAL, False),
+            'preview': (latest['metadata'].get(EXT_METADATA_ISPREVIEW, False) or
+                       latest['metadata'].get(EXT_METADATA_ISEXPERIMENTAL, False) or
+                       is_preview_from_semantic_version(latest['metadata']['version'])),
```

Comment:
The logic repeated in three places. I think it's better moved into one function.